index.js

```diff
@@ -10,10 +10,20 @@ var path = require('path');
 var ReplaceSource = require('webpack-core/lib/ReplaceSource');
 var util = require('./util');
 var WebIntegrityJsonpMainTemplatePlugin = require('./jmtp');
+var HtmlWebpackPlugin;
 
 // https://www.w3.org/TR/2016/REC-SRI-20160623/#cryptographic-hash-functions
 var standardHashFuncNames = ['sha256', 'sha384', 'sha512'];
 
+try {
```

Comment:
This plugin is a bit of an edge case because it plugs into HWP but also works without it. Is that common in your other plugins? I guess for dedicated plugins (that specify a peer dependency) it's ok to just use a straightforward `require`.

Reply:
Around 30% of the html-webpack-plugin use it as a peer dependency - but for all other cases a direct require should be the way to go

Reply:
Ah ok, so then it sounds like a helper package could be useful. Up to you. Out of curiosity, did you stop adding the hooks to `compilation.hooks` only because it's cleaner or is this also prescribed as best practice somewhere (by the Webpack team)?

Reply:
Ah there is already this one: https://github.com/tallesl/node-safe-require